api/TODO.adoc

```diff
@@ -0,0 +1,125 @@
+== Naming conventions
+
+Should we use plain names for classes/interfaces, using the package name to namespace them, or should we prefix them.
+
+`SyncCache`
+
+vs
+
+`org.infinispan.api.sync.SyncCache`
+
+== API dependencies
+
+The sync and async API must depend only on core JDK classes.
+The mutiny API has an optional dependency on mutiny.
+
+Should we depend on `javax.transaction` for the Transaction class or should we provide our own interface and a way to
```

Comment:
Or even just jakarta?

Reply:
I would create an abstraction to avoid depending on jakarta